src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -28,13 +28,45 @@ public static class Span
         public static ReadOnlyMemory<char> AsReadOnlyMemory(this string text)
         {
             if (text == null)
-            {
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.text);
-            }
 
             return new ReadOnlyMemory<char>(text, 0, text.Length);
         }
 
+        /// <summary>Creates a new <see cref="ReadOnlyMemory{T}"/> over the portion of the target string.</summary>
+        /// <param name="text">The target string.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown when <paramref name="text"/> is a null reference (Nothing in Visual Basic).</exception>
+        /// <exception cref="System.ArgumentOutOfRangeException">
+        /// Thrown when the specified <paramref name="start"/> index is not in range (&lt;0 or &gt;=text.Length).
```

Comment:
Comment vs code mismatch. `index is not in range (<0 or >=text.Length)`

The code throws if `start > text.Length`